pkg/testmachinery/testflow/node/node.go

```diff
@@ -151,7 +153,10 @@ func (n *Node) isRootNode() bool {
 	return n.inputSource != nil
 }
 
-func (n *Node) DetermineArtifacts() []argov1.Artifact {
+func (n *Node) GetOrDetermineArtifacts() []argov1.Artifact {
+	if artifacts != nil {
```

Comment:
where does artifacts come from?

please have a look at the test results

Reply:
done